client/packages/common/src/ui/layout/tables/material-react-table/useNonPaginatedMaterialTable.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import {
+  MRT_RowData,
+  MRT_TableOptions,
+  useMaterialReactTable,
+} from 'material-react-table';
+import {
+  CheckboxCheckedIcon,
+  CheckboxEmptyIcon,
+  CheckboxIndeterminateIcon,
+} from '@common/icons';
+
+interface NonPaginatedTableConfig<T extends MRT_RowData>
+  extends MRT_TableOptions<T> {
+  onRowClick?: (row: T) => void;
+  isLoading: boolean;
+}
+
+export const useBaseMaterialTable = <T extends MRT_RowData>({
```

Comment:
shared props didn't like working with the generic type, so instead using this as base, and passing in overrides from the paginated table hook

Reply:
Cool, should we put `useBaseMaterialReactTable` into a separate file?